### PR TITLE
b2: allow manual configuration of backblaze downloadUrl - fixes #2808

### LINKED ISSUE #2808: Request to have manual configuration of Backblaze downloadUrl
#### What is your current rclone version (output from `rclone version`)?
rclone v1.45
- os/arch: linux/amd64
- go version: go1.11.2

#### What problem are you are trying to solve?
When moving files, rclone downloads file and re-uploads for backblaze, which results in bandwidth usage during downloads.

#### How do you think rclone should be changed to solve that?
Recently cloudflare and backblaze made a bandwidth alliance, so if you could possible separate api url and ask for download url in backblaze configuration during setup, this problem can be solved by providing the download url for cloudflare enabled url rather than direct api.

### PR DESCRIPTION
**Pull request by trettet**
#### What is the purpose of this change?

Backblaze introduced free egress for B2 Cloud Storage when files are [downloaded through the Cloudflare network](https://www.backblaze.com/blog/backblaze-and-cloudflare-partner-to-provide-free-data-transfer/). This requires using a custom url or endpoint for downloads instead of the `downloadUrl` provided by `b2_authorize_account`.

#### Was the change discussed in an issue or in the forum before?

https://github.com/ncw/rclone/issues/2808

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ncw** [comment]
This looks great - thank you :-)

Presumably this will only work for public buckets?

I'll merge it now.

**Comment by trettet** [comment]
It should normally have an HTTP Error 401 when using it for private buckets, but with rclone, I don't really know why it doesn't. It even works for snapshots.

**Comment by ncw** [comment]
> So, yes, this would also work on private buckets without any further changes.

Interesting!

How do you get the Cloudflare endpoint you need?

**Comment by trettet** [comment]
There is a guide posted by Backblaze [here](https://help.backblaze.com/hc/en-us/articles/217666928-Using-Backblaze-B2-with-the-Cloudflare-CDN). This guide assumes that your `downloadUrl` does not change.

**Tl;dr:** You need to set a CNAME record on Cloudflare for your `downloadUrl` and make sure to have the orange cloudflare icon for that record so traffic passes through their network.
